### PR TITLE
Mongo db

### LINKED ISSUE #36: Connection to mongoDB
# User Story
As a **programmer**,
I want to **connect my project to mongoDB**,
so I can **add  and store my date in database**.

# Description

#Material

# Done

- [x] Setup mongoDB
- [x] Connect (!?)
- [x] (...?)

### PR DESCRIPTION
**Pull request by LebkuchenCJ**
Fix #36 

## Comments: 
- I finally connected to mongoDB and created my first routes.
- More routes will follow in a different PR, so the app is not yet fully functional
- Deployment errors will be fixed in a diffrent PR if occured

As a **programmer**,
I want to **connect my project to mongoDB**,
so I can **add and store my date in database**.

### CODE REVIEW

File: lib/cli.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+    const query = { name: "Cheese" };
+    const result = await collection.findOne(query);
+
+    console.log(result);
+  } finally {
+    await client.close();
+  }
+}
+run().catch(console.dir);
```
Comment (lmachens) [inline lib/cli.js]: Is this file only for test purpose? Do you still need it?

File: lib/models/list.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+const { ObjectID } = require("mongodb");
+const { getCollection } = require("../database");
+
+async function readLists() {
+  const listCollection = await getCollection("lists");
+  const lists = await listCollection.find({}).toArray();
+  if (!lists) {
+    return null;
+  }
```
Comment (lmachens) [inline lib/models/list.js:9]: When is `lists` falsy?

File: lib/routes/list.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
+const { Router } = require("express");
+const router = Router();
+
+const { readLists, readList, addList } = require("../models/list");
+
+router.get("/", async (request, response) => {
+  try {
+    const lists = await readLists();
+    if (!lists) {
+      response.status(404).send(`Lists not found`);
```
Comment (lmachens) [inline lib/routes/list.js]: If you do not return in your if-statement, the following code is executed too.

```suggestion
      return response.status(404).send(`Lists not found`);
```

File: package.json [hunk 4]
```diff
@@ -27,8 +27,11 @@
   },
   "homepage": "https://github.com/JonasImm/fullstack-template#readme",
   "dependencies": {
+    "dotenv": "^8.2.0",
     "express": "^4.17.1",
     "json-server": "^0.16.1",
+    "jsonwebtoken": "^8.5.1",
```
Comment (lmachens) [inline package.json]: I do not find any usage of `jsonwebtoken` -> it's not used and should not be in this pull request.

**Review by lmachens — COMMENTED** [review]
good job 👍

File: package.json (hunk 4 above)
Comment (LebkuchenCJ) [reply package.json]: Good point. Will fix it

File: lib/cli.js (hunk 1 above)
Comment (LebkuchenCJ) [reply lib/cli.js]: Forgot to delete it and will adapt asap

**Pull request merged**